worker/worker.go

```diff
@@ -29,6 +31,29 @@ type Worker struct {
 	QueueId        string
 }
 
+type TaskState uint8
+
+const (
+	TaskPending TaskState = iota
+	TaskRunning
+	TaskFinished
+	TaskFailed
+)
+
```

Comment:
Its a good practice has a string representation of enums, you can aim this by something like the following function:
```
func (ts TaskState) String() string {
	return [...]string{"TaskPending ", "TaskRunning", "TaskFinished", "TaskFailed"}[ts]
}
```